frontend/public/components/monitoring/dashboards/index.tsx

```diff
@@ -0,0 +1,253 @@
+  const items = _.zipObject(boards, boards);
+  return <Dropdown items={items} onChange={onChange} selectedKey={boards[0]} />;
+};
+
+const DurationDropdown: React.FC<DurationDropdownProps> = ({ items, onChange, selected }) => (
+  <Dropdown
+    buttonClassName="monitoring-dashboards-dropdown"
+    items={_.zipObject(items, items)}
+    onChange={(v: string) => onChange(parsePrometheusDuration(v))}
+    selectedKey={selected}
+  />
+);
+
+const Panel: React.FC<PanelProps> = ({ panel, pollInterval, timespan }) => {
+  const queries = _.map(panel.targets, 'expr').map(evaluateTemplate);
+  if (!queries.length) {
+    return null;
+  }
+
+  // If panel doesn't specify a span, try to get it from the gridPos or default to full width
+  let colSpan = panel.span;
+  if (!_.isNumber(colSpan)) {
+    colSpan = _.has(panel, 'gridPos.w') ? panel.gridPos.w / 2 : 12;
+  }
+
+  return (
+    <div className={`col-xs-${colSpan}`}>
+      <DashboardCard className="monitoring-dashboards-panel">
+        <DashboardCardHeader>
+          <DashboardCardTitle>{panel.title}</DashboardCardTitle>
+        </DashboardCardHeader>
+        <DashboardCardBody>
+          {panel.type === 'grafana-piechart-panel' && <BarChart query={queries[0]} />}
+          {panel.type === 'graph' && (
+            <Graph pollInterval={pollInterval} queries={queries} timespan={timespan} />
+          )}
+          {panel.type === 'row' && !_.isEmpty(panel.panels) && (
+            <div className="row">
+              {_.map(panel.panels, (p) => (
+                <Panel key={p.id} panel={p} pollInterval={pollInterval} timespan={timespan} />
+              ))}
+            </div>
+          )}
+          {panel.type === 'singlestat' && (
+            <SingleStat
+              decimals={panel.decimals}
+              format={panel.format}
+              pollInterval={pollInterval}
+              postfix={panel.postfix}
+              prefix={panel.prefix}
+              query={queries[0]}
+              units={panel.units}
+            />
+          )}
+          {panel.type === 'table' && <Table pollInterval={pollInterval} queries={queries} />}
+        </DashboardCardBody>
+      </DashboardCard>
+    </div>
+  );
+};
+
+const Board: React.FC<{ board: string; pollInterval: number; timespan: number }> = ({
+  board,
+  pollInterval,
+  timespan,
+}) => {
+  const [data, setData] = React.useState();
+  const [error, setError] = React.useState();
+
+  React.useEffect(() => {
+    const path = `${k8sBasePath}/api/v1/namespaces/openshift-monitoring/configmaps/grafana-dashboard-${board}`;
+    coFetchJSON(path)
+      .then((response) => {
+        const json = _.get(response, ['data', `${board}.json`]);
+        if (!json) {
+          setData(undefined);
+          setError('Dashboard definition JSON not found');
+        } else {
+          setData(JSON.parse(json));
+          setError(undefined);
+        }
+      })
+      .catch((err) => {
+        if (err.name !== 'AbortError') {
```

Comment:
I don't think this will even be thrown for `coFetchJSON` since you're not calling `useSafeFetch`

Reply:
Oops. I changed it to use `useSafeFetch` instead.